src/main/java/de/themorpheus/edu/taskservice/controller/TaskController.java

```diff
@@ -157,4 +162,19 @@ public Object getUserData(UUID userId) {
 		return this.taskRepository.findAllByAuthorId(userId);
 	}
 
+	public ControllerResult<GetMarkTaskAsDoneResponseDTO> markTaskAsDone(int taskId, Date date) {
+		Optional<TaskModel> optionalTask = this.taskRepository.getTaskByTaskId(taskId);
+		if (!optionalTask.isPresent()) return ControllerResult.of(Error.NOT_FOUND, Constants.Task.TaskDone.NAME_KEY);
+
+		TaskModel task = optionalTask.get();
```

Comment:
You need to write a query because you want to know if the exists in the same row

Reply:
furthermore use Constants.UserId.TEST_UUID

Reply:
query?

Reply:
method in the Repository